### ISSUE TITLE
Create new textures for double mapped ones

### ISSUE THREAD

**Issue by gilbertohasnofb** [c172p-team/c172p#1104]
Some textures have been mapped to more than one single object. This should be avoided as it breaks the AO and light maps. These objects should be mapped to copies of these textures.

The following is a list of items that will have to be dealt with at some point if we want light maps and good AO mapping:

![ant1a](https://user-images.githubusercontent.com/11069680/39964240-dc9a5fee-5645-11e8-97a1-8d5bf9504944.jpg)
![ant1](https://user-images.githubusercontent.com/11069680/39964246-fb62143a-5645-11e8-86d7-c8a67877a3ec.jpg)
![ant2](https://user-images.githubusercontent.com/11069680/39964248-0ed157b0-5646-11e8-94aa-523b857f28cb.jpg)
![ant3](https://user-images.githubusercontent.com/11069680/39964249-11bbe76a-5646-11e8-89bd-c59094e35850.jpg)
![ant4](https://user-images.githubusercontent.com/11069680/39964252-17056b42-5646-11e8-9485-f8e612d4d3d4.jpg)
![ant4a](https://user-images.githubusercontent.com/11069680/39964253-19059f48-5646-11e8-8447-52bdbe217ca8.jpg)

Below starts items that need their own texture space so the light maps will work properly. This first one is special as the interior metal ring on the glass (x2) should be mapped to the panel_parts.png texture so they can pick up the appropriate light for the light maps. But the outside rings (x2) on the glass should be mapped to fuselage.png do they aren't part of the interior and thus don't pick up in the light maps.
![ant5](https://user-images.githubusercontent.com/11069680/39964282-bbe26c0a-5646-11e8-8f69-b996a1d8810f.jpg)
![ant6](https://user-images.githubusercontent.com/11069680/39964286-dfb35e00-5646-11e8-8b46-4e5e89c75a46.jpg)

The window handles and the next images below of the door handles should be mapped to different textures for left and right doors and windows as different light sources will affect each one differently, so they each need their own texture space.
![ant7](https://user-images.githubusercontent.com/11069680/39964300-2ca19a38-5647-11e8-9f03-272cda459db2.jpg)

Any knobs as below that are mapped to the same texture space need to be remapped and provided their own separate texture space do the light maps work correctly..
![ant8](https://user-images.githubusercontent.com/11069680/39964308-6d742300-5647-11e8-9f58-2fabd2478def.jpg)

And last, every breaker needs it's own separate graphic, every switch needs it's own separate graphic . Upper right hand corner, that knob base needs it's own texture and space. Also look at the lower right (red circle around the dark texture), now this type of mapping will work if I remove any overlapping parts as each uv mapped object actually has its own space here. At least it will once I separate them by moving them apart a little bit.
![map1](https://user-images.githubusercontent.com/11069680/39964347-2cb28b4e-5648-11e8-93a5-790c3f36ccbb.jpg)

**Comment by gilbertohasnofb** [comment]
@wlbragg Because modifying textures for each livery, exporting and replacing them is a pain, here is the plan to fix this: I will first create new textures only for the default livery. This might involve some re-UV-mapping I'm afraid, since spacing suddenly has become critical. Once these are done and the re-UV-mapping is also done, we can bake new AO maps. Once that is done, I will then add the modifications to all liveries and apply the AO map to all of them as well. This means that for a short while all liveries except the default one will be broken in our branch, but this will be for a short time. Makes sense to you?

**Comment by wlbragg** [comment]
Current mapping of everything on interior.png. If you need me to decipher any of the mappings, let me know. We may have to add another texture sheet? Although there is plenty of white space with better organization and some of the large textures such as carpet and chassis I'm sure could be reworked to gain space back.
![int1](https://user-images.githubusercontent.com/11069680/39970620-3f95a9fa-56b4-11e8-836a-4fbae6f79f10.jpg)
![int2](https://user-images.githubusercontent.com/11069680/39970622-411287f8-56b4-11e8-9dc8-640b80aef3a1.jpg)

**Comment by wlbragg** [comment]
We have room on panel.png that we could potentially more something from interior.png over to if we need to.

**Comment by gilbertohasnofb** [comment]
@wlbragg Ok, I think we need to talk. Separating the interior will be an absolutely nightmare for me, given all the liveries and the number of objects involved... I just don't think it's worth just for the light map, in particular given the discussion we had [here](https://github.com/c172p-team/c172p-detailed/issues/438) and the fact that we already have the flashlight implemented. I also don't think that lowering the texture resolution in order to make them fit in the same space will justify a small 'special effect' if you know what I mean. This all have to do with the model we inherited and how it was first UV-mapped in the first place, of course. With a clean UV-map layout on a new project this would be the optimal solution (e.g. that's how I did the textures for the c182s), but given the amount of work we put during all these years we can't just press the reset button. I think we will just have to live with that and I _really_ hope you will understand my side on this one...

I would propose to only fix the textures that are 'wrongly' double mapped, e.g. using the metal bit of the throttle lever to map an antenna, etc. Then I would deal with the AOmaps manually, cleaning anything by hand (which is still much less pain than what I described above).

**Comment by gilbertohasnofb** [comment]
@wlbragg Alternatively, here is one idea: let's model only the roof lamp which is anyway the only thing that comes from factory to the model P. If we position this light source (which has two modes, red and white) exactly in the centre of the ceiling in between the two seats, then the lightmap will invariably be symmetrical on the doors and pilot and copilot seats. This way we can get away without needing to unduplicate those. If you agree that's feasible (and I super hope it is as the `interior.png` is the main problem here) then I would be up to unduplicate all stuff in the `panel_parts.png`, including breakers, switches, etc. I would also add new textures to `fuselage.png` for the antennas and metallic bits as discussed before. Is this route fine with you? If so I can start working on this now.

**Comment by gilbertohasnofb** [comment]
The only thing I would need from you is to map the copilot seat as a 'reflected' version of the pilot seat, is that doable in Blender? Then when creating the AO map and light maps you will need to select only one seat and only one door as not to count both on the final result.

**Comment by wlbragg** [comment]
@gilbertohasnofb that's fine! I can't say as I blame you. If it was just the one texture sheet this really wouldn't be that big of an ordeal. But to make that change throughout the entire livery, well, I get it!
No problem from me.
With that said I think we are doing enough with panel.png and panel_parts.png that we can get at least a couple of the light maps we want. The one that will probably be ruled out would be the "dome" light. However, depending on how directional they are, we may even be able to fudge it and still have some kind of imperfect dome light. I was also thinking about the ra-radiance effect. If we can use it to simulate the low energy light coming from the panel illumination, why can't we adapt that system to possibly simulate the added light coming from a semi-directional dome light. In other words, it may work well enough to have a dome light that is pointed mostly at the panel (we can do that with a light map now that we are at least separating that texture layout) and then add a little RA light to the doors and chassis when that dome light is on. It may work well enough. We'll just have to try it and experiment.

**Comment by wlbragg** [comment]
@gilbertohasnofb  we posted at the same time and I think we are indeed on the same page.

**Comment by gilbertohasnofb** [comment]
Fantastic. So I will start with the clean up of `panel_parts` and `fuselage`, please do take a look at my comment concerning the headrests and my comment above about flipping the mapping of the copilot seats. Once that's done we can move back to AO maps and then finally light maps.

**Comment by wlbragg** [comment]
>I would be up to unduplicate all stuff in the panel_parts.png, including breakers, switches, etc. I would also add new textures to fuselage.png for the antennas and metallic bits as discussed before. Is this route fine with you? If so I can start working on this now.

Yes, please do.

>then the lightmap will invariably be symmetrical on the doors and pilot and copilot seats

Yeah, should work.

>map the copilot seat as a 'reflected' version of the pilot seat, is that doable in Blender? 

I'm looking to see what you mean, I'll report back as soon as I have a look.

**Comment by wlbragg** [comment]
>map the copilot seat as a 'reflected' version of the pilot seat, is that doable in Blender?

I see what your saying, should be able to do that.

**Comment by gilbertohasnofb** [comment]
@wlbragg I am ready to push `panel_parts`, now I will work on `fuselage`. 

Ok, the panel_parts and fuselage textures are done for 4X-CHV livery. Note that:

* due to the symmetry of the light map, I did not bother with the door and window handle
* I found some more unused textures which freed even more space. 
* the breakers and the avionics switch were mapped to `/Models/Interior/Panel/Buttons/breakers/breakers.png`. I copied the contents of this texture to panel_parts and so breakers and the avionics switch must be remaped. I left the `breakers.png` file there in case you need a reference for the re-UV-mapping, but it must be deleted once this is done. See my messy reference below and let me know if anything is unclear.

![image](https://user-images.githubusercontent.com/5700795/39972065-46087c36-56ff-11e8-9708-6aecabcd9b5f.png)


* a lot of stuff had to be moved around I'm afraid... I am also adding a before and after for you. Before:

![image](https://user-images.githubusercontent.com/5700795/39972067-5901124e-56ff-11e8-8160-b4a419e7fda4.png)


After:

![image](https://user-images.githubusercontent.com/5700795/39972071-69ac4c8a-56ff-11e8-91ea-1abaa2c5af77.png)


Open both in different tabs and flick between them to see where things moved. I can help you if anything is still confusing, ok?

**Comment by gilbertohasnofb** [comment]
I'm done with `fuselage.png` as well, here are the new textures:

![image](https://user-images.githubusercontent.com/5700795/39972194-bc9fb81c-5701-11e8-8d04-37104ce2ea5c.png)

Let me know if anything is unclear.

**Comment by wlbragg** [comment]
@gilbertohasnofb it looks like we were both fooled. The breakers are mapped to their own texture sheet called breakers.png. I can remap them to the new ones you created and see wht they look like?

**Comment by gilbertohasnofb** [comment]
@wlbragg Fantastic, I will check it in the evening.

**Comment by wlbragg** [comment]
This part too.
![map](https://user-images.githubusercontent.com/11069680/40014268-b573a9bc-5775-11e8-8d34-43d7138b3132.jpg)

**Comment by gilbertohasnofb** [comment]
Ok, I will push a change in a minute. But can you tell me if this texture here (green arrow) is being used at all? 

![image](https://user-images.githubusercontent.com/5700795/40014303-1ae61ed8-57a8-11e8-9e15-4c723a9ddf9a.png)

**Comment by wlbragg** [comment]
Yes it is. No hurry, I'll be gone for awhile.

**Comment by gilbertohasnofb** [comment]
What is using that? The primer? 

Ok, I am pushing the fixed texture now, here is the texture for the illumination knob:

![image](https://user-images.githubusercontent.com/5700795/40014804-6edbae08-57a9-11e8-83d7-e07deffd7ca3.png)


And one more thing, I changed my mind about what @legoboyvdlp wrote, I think the side part of the pedestal could do with a re-UV-mapping, it looks messy and will influence how good the AO map will look like:

![fg](https://user-images.githubusercontent.com/5700795/40014641-f8b721f8-57a8-11e8-81b1-eac8f3b43ff9.jpg)

**Comment by wlbragg** [comment]
>What is using that? The primer?

Yes

Thank you for the new texture!

I'll re-map the pedestal at the same time as I re-map the throttle and light_control, actually I am doing that now..

**Comment by wlbragg** [comment]
@gilbertohasnofb this stuff should all be finished and I should be ready to make the final AO bakes.

**Comment by gilbertohasnofb** [comment]
Looks really good!

**Comment by gilbertohasnofb** [comment]
@wlbragg you missed the texture for the glareshield vent

**Comment by wlbragg** [comment]
>@wlbragg you missed the texture for the glareshield vent

@gilbertohasnofb I guess I got distracted and wanted to make sure this was corrected. I think it is as I have the glare shield vent in my local version that will eventually be pushed to the PR and we also have accounted for it in the AO maps.. But in all honesty, I don't remember what this was about. If it was the AO map or something else, as in it wasn't mapped on the aircraft glare shield?
I just want to make sure we didn't miss this.
